### PR TITLE
fix(shorebird_cli): make ShorebirdJavaMixin work on Windows

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Without this change, `shorebird release` fails with the error: 

```
✗ Exception: Failed to extract version name from app bundle: 'java' is not recognized as an internal or external command, operable program or batch file.
```

I'm not sure if this is the most elegant or correct fix, but it works. We'll likely want to reference what flutter_tool does to handle this.

I've verified this fix on my Windows 11 laptop and my M2 MBP. 

## Type of Change

- [ ] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/shorebird_java_mixin.dart [hunk 1]
```diff
@@ -8,6 +8,14 @@ import 'package:shorebird_cli/src/command.dart';
 /// Mixin on [ShorebirdCommand] which exposes methods
 /// for determining the java path.
 mixin ShorebirdJavaMixin on ShorebirdCommand {
+  String? getJavaExecutable([Platform platform = const LocalPlatform()]) {
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/shorebird_java_mixin.dart:11]: I’m confused. Is what’s on main working? I thought we verified main and these changes were just reorganization 🤔

File: packages/shorebird_cli/lib/src/shorebird_java_mixin.dart [hunk 2]
```diff
@@ -28,8 +36,8 @@ mixin ShorebirdJavaMixin on ShorebirdCommand {
     }
 
     final candidateLocations = [
-      p.join(androidStudioPath, 'jbr'),
-      p.join(androidStudioPath, 'jre'),
+      p.join(androidStudioPath, 'jbr', 'bin'),
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/shorebird_java_mixin.dart]: Are these changes needed? I didn’t change this logic and I thought we verified that main worked.

File: packages/shorebird_cli/lib/src/shorebird_java_mixin.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/shorebird_java_mixin.dart:11]: What's on `main` does work, but on `main`, this code is only used to set `JAVA_HOME` as an environment variable when we invoke gradle. For that gradle invocation, we have to specify the full path to the executable (see https://github.com/shorebirdtech/shorebird/blob/main/packages/shorebird_cli/lib/src/shorebird_flavor_mixin.dart#L17) like we're doing here.

File: packages/shorebird_cli/lib/src/shorebird_java_mixin.dart (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/shorebird_java_mixin.dart]: This was a mistake - just reverted it. These are the correct values for `JAVA_HOME`.

File: packages/shorebird_cli/lib/src/shorebird_java_mixin.dart (hunk 1 above)
Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/shorebird_java_mixin.dart:11]: My brain is shutting down for the night, but the tl;dr is:

1. Gradle needs the `JAVA_HOME` environment variable to be set. JAVA_HOME is a path to the root of a JDK installation, _not_ to the JDK's bin folder where the actual java executable lives.
2. `process.run` needs either a path to the executable or for the executable to be on the path.

I've renamed `getJavaPath` to make this distinction a bit clearer. We can chat more tomorrow in the very likely event that I'm not making sense here.

**Review by felangel — APPROVED** [review]
LGTM thanks so much for validating windows and catching this!
